imports/plugins/core/accounts/client/containers/addressBookContainer.js

```diff
@@ -0,0 +1,214 @@
+  Alerts.toast(errorMessage, "error");
+}
+
+const wrapComponent = (Comp) => (
+  class AddressBookContainer extends Component {
+    static propTypes = {
+      /**
+       * array of address objects
+       */
+      addressBook: PropTypes.arrayOf(PropTypes.shape({
+        _id: PropTypes.String,
+        fullName: PropTypes.String,
+        address1: PropTypes.String,
+        addresss2: PropTypes.String,
+        postal: PropTypes.String,
+        city: PropTypes.String,
+        region: PropTypes.String,
+        country: PropTypes.String,
+        phone: PropTypes.String,
+        isBillingDefault: PropTypes.Bool,
+        isShippingDefault: PropTypes.Bool,
+        isCommercal: PropTypes.Bool
+      })),
+      /**
+       * country options for select
+       */
+      countries: PropTypes.arrayOf(PropTypes.shape({
+        label: PropTypes.String,
+        value: PropTypes.String
+      })),
+      /**
+       *  Heading content for address book
+       */
+      heading: PropTypes.shape({
+        /**
+         * Heading title
+         */
+        defaultValue: PropTypes.String,
+        /**
+         * i18nKey for heading title
+         */
+        i18nKey: PropTypes.String,
+        /**
+         * If in checkout view, addressbook checkout step position and icon className
+         */
+        checkout: PropTypes.shape({
+          icon: PropTypes.String,
+          position: PropTypes.Number
+        })
+      }),
+      /**
+       * regions by county
+       */
+      regionsByCountry: PropTypes.shape({
+        countryCode: PropTypes.arrayOf(PropTypes.shape({
+          label: PropTypes.String,
+          value: PropTypes.String
+        }))
+      })
+    }
+
+    render() {
+      return (
+        <Comp
+          {...this.props}
+          addAddress={addAddress}
+          updateAddress={updateAddress}
+          removeAddress={removeAddress}
+          onError={onError}
+        />
+      );
+    }
+  }
+);
+
+function composer(props, onData) {
+  const account = Collections.Accounts.findOne({ _id: Meteor.userId() });
+  const { addressBook } = account.profile;
+  const countries = Countries.find().fetch();
+  const template = Template.instance();
+  const { data } = template || { data: undefined };
+  const { heading: templateHeading } = data || { heading: undefined };
+  const shop = Collections.Shops.findOne();
+  const shopCountries = shop.locales.countries;
+
+  let regionsByCountry;
+  Object.keys(shopCountries).forEach((key) => {
+    const { states } = shopCountries[key] || undefined;
+    const regions = [];
+    const country = {};
+    if (states) {
+      // states is an object that needs to be convered
+      // to an array of region labels and values
+      Object.keys(states).forEach((i) => {
+        regions.push({
+          label: states[i].name,
+          value: i
+        });
+      });
+    }
+    country[key] = regions;
+    regionsByCountry = { ...regionsByCountry, ...country };
+  });
```

Comment:
This will create a new map every loop, which is inefficient. Since you're just building the `regionsByCountry` map right now, you don't have to worry about mutating it causing problems. So you can do:

```js
const regionsByCountry = {};
```

And then skip the `country` variable and instead directly do `regionsByCountry[key] = regions;`